more Dual maths function implementations

claudeha opened the pull request:
atan(y,x) exp log sinh cosh tanh asin acos atan asinh acosh atanh

claudeha committed:
more Dual maths function implementations

atan(y,x) exp log sinh cosh tanh
asin acos atan asinh acosh atanh

lycium reviewed (commented):
Hi Claude, thanks for your submission!

Two important things I noticed: `const Dual<real_type, vars> & r;` in atan should be without const and reference (&), and instead of e.g. sqrt we should use std::sqrt so it uses the single precision one when appropriate (instead of always double).

I can fix this in another commit when I'm properly up :)

claudeha committed:
fix typo

claudeha committed:
rename atan(y, x) to atan2(y, x)

claudeha committed:
use std:: for maths functions

claudeha commented:
On 06/03/2021 12:08, Thomas Ludwig wrote:
> Two important things I noticed: |const Dual<real_type, vars> & r;| 
> should be without reference (&), and instead of e.g. sqrt we should 
> use std::sqrt so it uses the single precision one when appropriate 
> (instead of always double).
I fixed these in a couple more commits.  Also renamed atan of two 
arguments to atan2, not sure what the C++ spec is but I guess it follows C?

lycium commented:
> I fixed these in a couple more commits.  Also renamed atan of two arguments to atan2, not sure what the C++ spec is but I guess it follows C?

Yep this is perfect now, thanks again for the great addition!
